raiden/raiden_service.py

```diff
@@ -466,20 +465,47 @@ def add_pending_greenlet(self, greenlet: gevent.Greenlet):
     def __repr__(self):
         return '<{} {}>'.format(self.__class__.__name__, pex(self.address))
 
-    def start_neighbours_healthcheck(self, chain_state: ChainState):
+    def _initialize_transport(self, chain_state: ChainState):
+        """ Initialize the transport and related facilities.
+
+        Note:
+            The transport must not be started before the node has caught up
+            with the blockchain through `AlarmTask.first_run()`. This
+            synchronization includes the on-chain channel state and is
+            necessary to reject new messages for closed channels.
+        """
+        assert self.alarm.is_primed(), 'AlarmTask not primed. node:{self!r}'
+        assert not self.ready_to_process_events
+
+        self._initialize_messages_queues(chain_state)
+        self._initialize_whitelists(chain_state)
+
+        self.ready_to_process_events = True
```

Comment:
I wanted to have everything transport related in a method, but then this guy happened, either we leave it here or split the initialization from side-effects.